### ISSUE TITLE
Visual Studio Debugger closes on file upload

### ISSUE THREAD

**Issue by hartmair** [brave/brave-browser#21364]
## Description 
When I upload a file debugging ASP.NET Core apps with the Brave Browser having the Visual Studio Debugger attached, the browser closes.

## Steps to Reproduce

See https://github.com/dotnet/aspnetcore/issues/28547

   1. Create a new .NET 6 Blazor Server App
   2. Open `Index.razor` and paste the following at the end: 
   ```razor
   <InputFile OnChange="@UploadFile" />
@code{
    private void UploadFile() { }
}
```
   3. Start Debugging using Brave Browser
   4. Select a file. The server application will close after the select file dialog is gone. 

## Actual result:
Browser closes on file upload

## Expected result:
Browser continues running after file upload

## Reproduces how often: 
Easily reproduced

## Brave version (brave://version info)
1.35.103 Chromium: 98.0.4758.102 (Offizieller Build) (64-Bit)

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? Sometimes
- Does the issue resolve itself when disabling Brave Rewards? No
- Is the issue reproducible on the latest version of Chrome? No

## Miscellaneous Information:
See https://github.com/dotnet/aspnetcore/issues/28547#issuecomment-1044958222
See https://stackoverflow.com/a/59611796/1220972
See https://stackoverflow.com/a/58059150/1220972

**Comment by rebron** [comment]
cc: @ryanbr

**Comment by ryanbr** [comment]
If you disable ads/trackers in shields, does it help. If so, copy/paste the blocked items here @hartmair

**Comment by hartmair** [comment]
@ryanbr 

> If you disable ads/trackers in shields, does it help. If so, copy/paste the blocked items here

No, doesn't help for me, although some users on Stackoverflow reported otherwise. (I updated the description accordingly)

**Comment by Atif-Rehmat** [comment]
I m facing the same issue, although I m not using Blazor. I m using Angular 13 and browser is Brave. 
Framework: .NET 6
Visual Studio: Professional 2022
Windows: 11 Pro version 21H2

**Comment by shaldev** [comment]
I disabled browser link in vs and this stopped happening.

**Comment by shutupandshave** [comment]
I also have this issue, I dont have browser link active.

**Comment by utilsites** [comment]
Also worked for me the solution from @NMESteve, thanks!

It seems that visual studio is monitoring the brave browser window that was opened by the visual studio debugging session, and when the user tries to upload a file in the website, after closing the upload file window, it seems that visual studio interprets like the whole browser window/tab was closed and then stops the debugger and process.
This occurs even without BrowserLink enabled. 
Don't know if the problem is from the Browser side by some implemented interface in a wrong way or from Visual Studio side. Going try to report to Visual Studio side too.

> This seemed to be the issue for me. ![image](https://private-user-images.githubusercontent.com/4820386/289633863-64976c79-776b-4f96-94e5-a2522ea5727a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUzNDM1MzEsIm5iZiI6MTcwNTM0MzIzMSwicGF0aCI6Ii80ODIwMzg2LzI4OTYzMzg2My02NDk3NmM3OS03NzZiLTRmOTYtOTRlNS1hMjUyMmVhNTcyN2EucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDExNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMTVUMTgyNzExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDU3Zjk5OGQyMTQ2MWQ3MWJmZTg3NjZhODEwNGQ5OGY1ZmE5NDdmNWEyMGJkOGVlZDRhYjcyODk1ZGI1NWRhNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.0-Ye-4KQrAPGjCIERSzuO1ma2hHzsDfvQSGaYdQ1aRc)

**Comment by utilsites** [comment]
Related thread from Visual Studio side: https://developercommunity.visualstudio.com/t/Visual-Studio-debugger-stops-when-debugg/10561711

**Comment by Erziok** [comment]
> This seemed to be the issue for me. ![image](https://github.com/brave/brave-browser/assets/4820386/64976c79-776b-4f96-94e5-a2522ea5727a)

It worked for me, thanks!

**Comment by vijithkrthe** [comment]
Brave browser still have this issue , Stop debugger when browser window is closed, close browser when debugging
stops when I Disable this option the problem not occurred however I'm not sure is it a good idea to do that is there is any problem on doing this that's still unknown to me

**Comment by rsk-paul** [comment]
An alternate fix is to set: `"launchBrowser": false` in `launchSettings.json`

**Comment by kimjamia** [comment]
Even if your default browser is Brave, you can use a different browser when debugging: https://stackoverflow.com/a/71497307
